File: database/seeds/category.go
package main

import (
	"fmt"

	"github.com/uyupun/regret/database"
	"github.com/uyupun/regret/models"
)

func SeedCategory() {
	category := models.Category{
		ID:        1,
		ServiceId: 1,
		Name:      "test",
		NameJa:    "test",
	}

	db, err := database.ConnectGorm()
	if err != nil {
		panic(err)
	}
	db.Create(&category)

	fmt.Println("Category seeding succeeded!")
}

File: cpanel/src/models/category.ts
export interface Category {
  id: number;
  service_id: number;
  name: string;
  name_ja: string;
};

export const initCategory = () => {
  return {
    id: 0,
    service_id: 0,
    name: '',
    name_ja: ''
  };
};

export interface AddCategory {
  service_id: number;
  name: string;
  name_ja: string;
};

export const initAddCategory = () => {
  return {
    service_id: 0,
    name: '',
    name_ja: ''
  };
};
